### PR TITLE
refactor: adjust DSL indentation in tests inside package server

### LINKED ISSUE #1302: Update `openfga/language` Go module for DSL transformer/parser and refactor usages
### Checklist

- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Describe the problem you'd like to have solved

A lot of our tests today use the openfga/language parser for Go to make it easier to write tests using our DSL syntax instead of the JSON model syntax. You can see a few examples of this here:

https://github.com/openfga/openfga/blob/main/internal/graph/check_test.go#L43-L55
https://github.com/openfga/openfga/blob/main/internal/graph/cached_resolver_test.go#L449-L469
https://github.com/openfga/openfga/blob/main/pkg/server/server_test.go#L182-L188

A more recent commit/release of the parser supports whitespace better, and so we can refactor these usages to have better usage of whitespace in the test files. The existing tests today had to swallow whitespace to avoid parser errors, and so they read quite cryptically relative to the other whitespace in the file.

### Describe the ideal solution

Upgrade the parser and fix any of the whitespace in the tests that use the parser to parse a DSL model string.

### Alternatives and current workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 00chorch**
Fixes #1302 for package server

## Description

Now that openfga/language parser has better support for whitespaces, refactor to
have better readability in the test files.

## References

Issue #1302 

## Review Checklist
- [X] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [X] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/server/commands/list_objects_test.go [hunk 1]
```diff
@@ -46,14 +46,14 @@ func TestListObjectsDispatchCount(t *testing.T) {
 		{
 			name: "test_direct_relation",
 			model: `model
-			schema 1.1
+				schema 1.1
```
Comment (jon-whit) [inline pkg/server/commands/list_objects_test.go:49]: This is really nested - can you put the `model` clause on a newline below and start everything inline from there. For example,

```go
model: `
model
  schema 1.1

type user
type document
  relations
    define viewer: [user]
 `,
```

Comment (00chorch) [reply pkg/server/commands/list_objects_test.go:49]: Thanks for your feedback @jon-whit 
I wrote a script for this. I'll adjust it to move the content after ` to be in a new line aligned at column 0

Here's a new example with the expected output
~~~
	model := testutils.MustTransformDSLToProtoWithID(`
model
schema 1.1

type user

type document
  relations
    define viewer: [user, document#viewer] or editor
    define editor: [user, document#viewer]`)
~~~

Once approved, I can take care of other packages by just re-running the script

File: pkg/server/commands/list_objects_test.go [hunk 2]
```diff
@@ -45,15 +45,16 @@ func TestListObjectsDispatchCount(t *testing.T) {
...
+type user
 
-			type folder
-				relations
-					define viewer: [user] 
-			`,
+type folder
+	relations
+		define viewer: [user] 
+	`,
```
Comment (miparnisari) [inline pkg/server/commands/list_objects_test.go:57]: I liked it better the way it was before 
image

File: pkg/server/commands/write_test.go [hunk 3]
```diff
@@ -238,13 +238,14 @@ func TestTransactionalWriteFailedError(t *testing.T) {
...
+schema 1.1
 type user
 
 type document
-  relations
-	define viewer: [user]`), nil)
+	relations
+		define viewer: [user]`), nil)
 
 	mockDatastore.EXPECT().
```
Comment (miparnisari) [inline pkg/server/commands/write_test.go:250]: I like the model definition to be aligned vertically with whatever comes before it

image

Comment (00chorch) [reply pkg/server/commands/write_test.go:250]: FYI: accidentally closed PR. I will resubmit respecting previous lines indentation. ETA today/tomorrow

**Pull request closed without merging**